registry/registry.go

```diff
@@ -723,11 +737,89 @@ func tarReader(reader io.Reader, contentType string) (*tar.Reader, error) {
 	return tar.NewReader(reader), nil
 }
 
+// CheckVersion controls the matching versions between retrieved tarball and
+// options
+func (t *Tarball) CheckVersion(expectedVersion string) (bool, error) {
+	tbManifestVersion := t.Manifest.Version
+	tbPackageVersion := t.PackageVersion
+
+	var match bool
+	var errm error
+
+	if tbManifestVersion != "" {
+		match = VersionMatch(expectedVersion, tbManifestVersion)
+	}
+	if !match {
+		errm = multierror.Append(errm,
+			fmt.Errorf("%q field does not match (%q != %q)",
+				"version", expectedVersion, tbManifestVersion))
+	}
+
+	if tbPackageVersion != "" {
+		match = VersionMatch(expectedVersion, tbPackageVersion)
+		if !match {
+			errm = multierror.Append(errm,
+				fmt.Errorf("\"version\" from package.json (%q != %q)",
+					expectedVersion, tbPackageVersion))
+		}
+	}
+
+	if errm != nil {
+		err := errshttp.NewError(http.StatusUnprocessableEntity,
+			"Content of the manifest does not match: %s", errm)
+		return false, err
+	}
+
+	return true, nil
+}
+
+// CheckEditor validates a tarball manifest editor
+func (t *Tarball) CheckEditor() (bool, error) {
+	var errm error
+	editorName := t.Manifest.Editor
+
+	if editorName == "" {
+		errm = multierror.Append(errm,
```

Comment:
Why do you a multierror here?

Reply:
I tried to preserve the existing multi-error logic, but a classic error-check  is fine with me